text/0004-configurational-dependencies.md

```diff
@@ -0,0 +1,57 @@
+# Configurational dependencies
+
+## Summary
+
+A new type of dependencies for installing configuration and hooks.
+
+## Motivation
+
+We want to make it possible to share some configurations between projects. Storing the configuration in regular dependencies is not an option as we might need the configuration during installation of the "regular" dependencies ("dependencies", "devDependencies", and "optionalDependencies"). Hence, we need to install these configurational dependencies before other types of dependencies.
+
+Some examples of usage:
+
+* Installing hooks used by [`.pnpmfile.cjs`](https://pnpm.io/pnpmfile).
+* Installing the list of dependencies that are allowed to be built (the [pnpm.onlyBuiltDependenciesFile](https://pnpm.io/package_json#pnpmonlybuiltdependenciesfile)).
+* Installing [catalogs](https://pnpm.io/catalogs).
+* Loading patch file.
+
+## Detailed Explanation
+
+There will be a new field in `package.json` called `pnpm.configDependencies`. For example:
+
+```json
+{
+  "name": "my-pkg",
+  "version": "0.0.0",
+  "pnpm": {
+    "configDependencies": {
+      "my-configs": "1.0.0+sha512-30iZtAPgz+LTIYoeivqYo853f02jBYSd5uGnGpkFV0M3xOt9aN73erkgYAmZU43x4VfqcnLxW9Kpg3R5LC4YYw=="
+    }
+  }
+}
+```
+
+These new type of "configurational" dependencies will be npm packages with a lot of limitations:
+
+* They won't have any dependencies. Even if they will have dependencies, pnpm will ignore them during installation.
+* They will not have lifecycle scripts.
+* They will only be installable via exact versions.
+
+These dependencies will be installed into a new directory (name to be decided), not into `node_modules`.
```

Comment:
That's a good idea.

Reply:
Yeah, using node_modules could have advantages but would also require additional changes in the existing installation code. Unless, we use a dot directory inside node_modules, like `node_modules/.pnpm-config`. Or `node_modules/.pnpm/config`.

Reply:
I do like the idea of `node_modules/.pnpm-config` as an analogue to the existing `node_modules/.pnpm` dir. I feel like that makes it clearer that these are NPM packages, but live under a namespace isolated from the existing packages in `node_modules/.pnpm`.